### PR TITLE
Drop mock dependency

### PR DESCRIPTION
**Pull request by GallowayJ**
Dropped mock as a dependency to resolve issue #6665 by changing imports of mock to unittest.mock and removing mock from requirements.txt.

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ This version has two major changes:
 * It updates our default docker images to stable=5.0 and latest=6.0.
 * It changes our PR builder domain to `readthedocs.build`
 
+* `@gallowayj <https://github.com/gallowayj>`__: Drop mock as a dependency (`#6665 <https://github.com/readthedocs/readthedocs.org/pull/6665>`__)
```
Comment (saadmk11) [inline CHANGELOG.rst]: This is actually auto generated. So, I think we don't need to add this here.

Comment (GallowayJ) [reply CHANGELOG.rst]: @saadmk11 Sorry, should I cancel the pull requestion and resubmit a new pull request with the changes add to CHANGELOG.rst?

Comment (saadmk11) [reply CHANGELOG.rst]: No. Just remove this line and push it to the branch and the PR will be updated

Comment (GallowayJ) [reply CHANGELOG.rst]: @saadmk11 Thanks, I've removed that line and pushed the changes back to the branch.

**Pull request merged**